src/index.js

```diff
@@ -83,8 +83,11 @@ class SWPrecacheWebpackPlugin {
       }
 
       // add hash to importScripts
+      const publicPath = compiler.options.output.publicPath || '';
       const scripts = this.options.importScripts || [];
-      const importScripts = scripts.map(f => f.replace(/\[hash\]/g, stats.hash));
+      const importScripts = scripts
+        .map(f => f.replace(/\[hash\]/g, stats.hash))
+        .map(f => path.join(publicPath, f));
```

Comment:
@Aweary I found that node's `path` will remove duplicate`/`. So when a public path of `http://cdn.com/` is provided, it is converted to `http:/cdn.com/` (notice only one slash).
I'm changing this to: `.map(f => url.resolve(publicPath, f));`
hopefully that won't break anything for you. I'll do it in a minor release just in case it breaks something.